packages/components/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,186 @@
+
+  const handleRowClick = useCallback(
+    (row: Row<T>) => () => {
+      if (onRowClick == undefined) return;
+      onRowClick(row);
+    },
+    [onRowClick],
+  );
+
+  const table = useReactTable(tableSettings);
+  return (
+    <div className={styles.dataTableContainer}>
+      <div className={styles.tableContainer} style={{ height }}>
+        <table className={tableClasses}>
+          <thead className={stickyClass}>
+            {table.getHeaderGroups().map(headerGroup => (
+              <tr key={headerGroup.id}>
+                {headerGroup.headers.map(header => {
+                  return (
+                    <th
+                      key={header.id}
+                      colSpan={header.colSpan}
+                      className={
+                        sorting && header.column.getCanSort()
+                          ? classNames(styles.sortableColumn, {
+                              [styles.pinFirstHeaderSortable]: !!onRowClick,
+                            })
+                          : ""
+                      }
+                      onClick={
+                        sorting
+                          ? header.column.getToggleSortingHandler()
+                          : undefined
+                      }
+                    >
+                      {header.isPlaceholder ? null : (
+                        <div>
+                          {flexRender(
+                            header.column.columnDef.header,
+                            header.getContext(),
+                          )}
+                          {{
+                            asc: <Icon name="longArrowUp" color="green" />,
+                            desc: <Icon name="longArrowDown" color="green" />,
+                          }[header.column.getIsSorted() as string] ?? (
+                            <svg
+                              style={{
+                                width: "var(--space-large)",
+                                height: "var(--space-large)",
+                              }}
+                            />
+                          )}
+                        </div>
+                      )}
+                    </th>
+                  );
+                })}
+              </tr>
+            ))}
+          </thead>
+          <tbody>
+            {table.getRowModel().rows.map(row => {
+              return (
+                <tr
+                  key={row.id}
+                  onClick={handleRowClick(row)}
+                  className={bodyRowClasses}
+                >
+                  {row.getVisibleCells().map(cell => {
+                    return (
+                      <td key={cell.id}>
+                        {flexRender(
+                          cell.column.columnDef.cell,
+                          cell.getContext(),
+                        )}
+                      </td>
+                    );
+                  })}
+                </tr>
+              );
+            })}
+          </tbody>
```

Comment:
As we have the Footer component, is it makes sense to turn this part into a `Body` component?

Reply:
✅